module/VuFindTheme/src/VuFindTheme/ResourceContainer.php

```diff
@@ -130,11 +130,102 @@ public function addCss($css)
     public function addJs($js)
     {
         if (!is_array($js) && !is_a($js, 'Traversable')) {
-            $js = [$js];
+            $this->addJsEntry($js);
+        } elseif (isset($js['file'])) {
+            $this->addJsEntry($js);
+        } elseif (isset($js[0])) {
+            foreach ($js as $current) {
+                $this->addJsEntry($current);
+            }
+        } elseif ($js == []) {
+            return;
+        } else {
+            trigger_error("Invalid JS entry format: " . print_r($js, true));
+        }
+    }
+
+    /**
+     * Helper function for adding a Javascript file.
+     *
+     * @param string|array $jsEntry Entry to add, either as string with path
+     * or array with additional properties.
+     *
+     * @return void
+     */
+    protected function addJsEntry($jsEntry)
+    {
+        if (!is_array($jsEntry)) {
+            $parts = $this->parseSetting($jsEntry);
+            if (count($parts) == 1) {
+                $jsEntry = ['file' => $jsEntry];
+            } else {
+                $jsEntry = [
+                    'file' => $parts[0],
+                    'attributes' => ['conditional' => trim($parts[1])],
+                ];
+            }
+        } elseif (isset($jsEntry['priority']) && isset($jsEntry['load_after'])) {
+            throw new \Exception(
```

Comment:
It would be a good idea to add a small unit test to cover this case; you can use `$this->expectException(\Exception::class);` and `$this->expectExceptionMessage('...the message...');` at the beginning of the test case prior to triggering the exception. That way we can make sure this edge case doesn't break in the future (and we'll be reminded to test it properly if we ever change the behavior).

Reply:
The test case has been added, however it is not shown automatically as "outdated" because the change has been made in a different line.

Reply:
Thanks! You can feel free to mark conversations as resolved in situations like this. :-)

Reply:
Ok, i will 😉